pkg/cloudprovider/providers/gce/gce_disks.go

```diff
@@ -809,6 +862,48 @@ func (gce *GCECloud) DeleteDisk(diskToDelete string) error {
 	return err
 }
 
+func (gce *GCECloud) ResizeDisk(diskToResize string, oldSize resource.Quantity, newSize resource.Quantity) (resource.Quantity, error) {
+	disk, err := gce.GetDiskByNameUnknownZone(diskToResize)
+	if err != nil {
+		return oldSize, err
+	}
+
+	requestBytes := newSize.Value()
+	// GCE resizes in chunks of GBs (not GiB)
+	requestGB := volume.RoundUpSize(requestBytes, 1000*1000*1000)
+	newSizeQuant := resource.MustParse(fmt.Sprintf("%dG", requestGB))
+
+	// If disk is already of size equal or greater than requested size, we simply return
+	if disk.SizeGb >= requestGB {
+		return newSizeQuant, nil
+	}
+
+	var mc *metricContext
+
+	switch zoneInfo := disk.ZoneInfo.(type) {
+	case singleZone:
+		mc = newDiskMetricContextZonal("resize", disk.Region, zoneInfo.zone)
+		resizeOp, err := gce.manager.ResizeDiskOnCloudProvider(disk, requestGB, zoneInfo.zone)
+
+		if err != nil {
+			return oldSize, mc.Observe(err)
+		}
+		return newSizeQuant, gce.manager.WaitForZoneOp(resizeOp, zoneInfo.zone, mc)
```

Comment:
To stay consistent, `oldSize` should be returned if `WaitForZoneOp()` returns an error.

If resize hits the max quota and errors out, it's safer to back off to the oldSize imo.

Reply:
agreed. will do

Reply:
fixed